monix-tail/shared/src/main/scala/monix/tail/PullStrategy.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail
+
+sealed abstract class PullStrategy extends Product with Serializable
```

Comment:
I would also name it `ReactivePullStrategy` or something like that, because it's very much related to the `reactivestreams.org` protocol. If we'll add another `Introspective` strategy or something, it will only make sense via this protocol integration.